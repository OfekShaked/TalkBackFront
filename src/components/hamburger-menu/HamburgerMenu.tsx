import React from 'react';
import { Drawer,IconButton,Divider,List,ListItem,ListItemIcon,ListItemText} from '@material-ui/core';
import {ChevronLeft,ChevronRight,AccountCircle,Add,Launch,ExitToApp} from '@material-ui/icons';
import useStyles from './HamburgerStyle';
import { useTheme } from '@material-ui/core';
import {NavLink} from 'react-router-dom';


const HamburgerMenu = (props:any) =>{
  const theme = useTheme();
  const classes = useStyles(theme);
  
    return (
      <Drawer
      className={classes.drawer}
      variant="persistent"
      anchor="left"
      open={props.open}
      classes={{
        paper: classes.drawerPaper,
      }}
    >
      <div className={classes.drawerHeader}>
        <IconButton onClick={props.handleDrawerClose}>
          {theme.direction === 'ltr' ? <ChevronLeft /> : <ChevronRight />}
        </IconButton>
      </div>
      <Divider />
      <List>
        {!props.isLoggedIn ? (
          <>
          <ListItem button
          key="Sign-In"
          component={NavLink} to="/signin">
          <ListItemIcon><AccountCircle></AccountCircle></ListItemIcon>
          <ListItemText primary="Sign-In"></ListItemText>
        </ListItem>
        <ListItem button
          key="Register"
          component={NavLink} to="/register">
          <ListItemIcon><Add></Add></ListItemIcon>
          <ListItemText primary="Register"></ListItemText>
        </ListItem>
        </>):<>
        <ListItem button
          key="Open"
          component={NavLink} to="/contact">
          <ListItemIcon><Launch></Launch></ListItemIcon>
          <ListItemText primary="Open"></ListItemText>
        </ListItem><ListItem button
          key="Logout"
          component={NavLink} to="/Logout">
          <ListItemIcon><ExitToApp></ExitToApp></ListItemIcon>
          <ListItemText primary="Logout"></ListItemText>
        </ListItem></>}

      </List>
    </Drawer>
    );
}
export default HamburgerMenu;