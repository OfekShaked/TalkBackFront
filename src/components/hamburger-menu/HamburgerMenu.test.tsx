import React from 'react';
import renderer from 'react-test-renderer';
import HamburgerMenu from './HamburgerMenu';
import SocketMock from 'socket.io-mock';
import { act } from 'react-dom/test-utils';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';
import { ListItemText, ListItem } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';

configure({ adapter: new Adapter() });

const handleClose = jest.fn();

describe('hamburger menu tests', () => {
    it('renders hamburger menu component', async () => {
        const component = renderer.create(
            <Router>
                <HamburgerMenu open={true} handleDrawerClose={handleClose} isLoggedIn={true}></HamburgerMenu>
            </Router>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should have 2 links for logged in user", async () => {
        const component = mount(
            <Router>
                <HamburgerMenu open={true} handleDrawerClose={handleClose} isLoggedIn={true}></HamburgerMenu>
            </Router>
        );
        const links = component.find(ListItem);
        expect(links.length).toBe(2);
    });
    it("should render logged in user actions", async () => {
        const component = mount(
            <Router>
                <HamburgerMenu open={true} handleDrawerClose={handleClose} isLoggedIn={true}></HamburgerMenu>
            </Router>
        );
        const links = component.find(ListItemText);
        expect(links.at(0).props().primary).toBe("Open");
        expect(links.at(1).props().primary).toBe("Logout");
    });
    it("should have 2 links for guests", async () => {
        const component = mount(
            <Router>
                <HamburgerMenu open={true} handleDrawerClose={handleClose} isLoggedIn={false}></HamburgerMenu>
            </Router>
        );
        const links = component.find(ListItem);
        expect(links.length).toBe(2);
    });
    it("should render guests actions", async () => {
        const component = mount(
            <Router>
                <HamburgerMenu open={true} handleDrawerClose={handleClose} isLoggedIn={false}></HamburgerMenu>
            </Router>
        );
        const links = component.find(ListItemText);
        expect(links.at(0).props().primary).toBe("Sign-In");
        expect(links.at(1).props().primary).toBe("Register");
    });
})

