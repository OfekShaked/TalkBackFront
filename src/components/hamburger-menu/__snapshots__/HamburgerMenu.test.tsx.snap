// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hamburger menu tests renders hamburger menu component 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
>
  <div
    className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <div
      className="makeStyles-drawerHeader-3"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
      </button>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <ul
      className="MuiList-root MuiList-padding"
    >
      <a
        aria-current={null}
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        href="/contact"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Open
          </span>
        </div>
      </a>
      <a
        aria-current={null}
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        href="/Logout"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Logout
          </span>
        </div>
      </a>
    </ul>
  </div>
</div>
`;
